### PR TITLE
[MAMAEdu-81] Send email after enroll through API

### PR DESCRIPTION
**Pull request by dyudyunov**
**Youtrack:**
https://youtrack.raccoongang.com/issue/MAMAEdu-81

**Notes:**
- emails are sending translated to 'ru' (`language_code` setting)
- merge after #2066

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: common/djangoapps/enrollment/views.py [hunk 1]
```diff
@@ -648,6 +657,17 @@ def post(self, request):
                     is_active=is_active,
                     enrollment_attributes=enrollment_attributes
                 )
+                # try:
```
Comment (OPersian) [inline common/djangoapps/enrollment/views.py:660]: shouldn't it be removed?

Comment (dyudyunov) [reply common/djangoapps/enrollment/views.py:660]: OMG

File: common/djangoapps/enrollment/tests/test_views.py [hunk 2]
```diff
@@ -47,14 +47,15 @@ def assert_enrollment_status(
...
+            email=None,
             expected_status=status.HTTP_200_OK,
             email_opt_in=None,
             as_server=False,
             mode=CourseMode.DEFAULT_MODE_SLUG,
             is_active=None,
             enrollment_attributes=None,
             min_mongo_calls=0,
-            max_mongo_calls=0,
+            max_mongo_calls=1,  # see student.models.py::1235
```
Comment (OPersian) [inline common/djangoapps/enrollment/tests/test_views.py:58]: Probably, a more meaningful comment should be left here. I mean, what if a`student.models.py` changes and the line number will be different.

File: common/djangoapps/enrollment/views.py [hunk 3]
```diff
@@ -555,12 +560,16 @@ def post(self, request):
...
+                user = User.objects.get(username=username)
+            else:
+                user = User.objects.get(email=email)
+                username = user.username
         except ObjectDoesNotExist:
             return Response(
                 status=status.HTTP_406_NOT_ACCEPTABLE,
                 data={
-                    'message': u'The user {} does not exist.'.format(username)
+                    'message': u'The user {} does not exist.'.format(email if email else username)
```
Comment (OPersian) [inline common/djangoapps/enrollment/views.py:572]: `email or username`?

File: common/djangoapps/enrollment/tests/test_views.py (hunk 2 above)
Comment (dyudyunov) [reply common/djangoapps/enrollment/tests/test_views.py:58]: I want to remove that comment at all

Comment (OPersian) [reply common/djangoapps/enrollment/tests/test_views.py:58]: I guess I left a similar comment somewhere; sorry if I missed your reply

Comment (OPersian) [reply common/djangoapps/enrollment/tests/test_views.py:58]: Ah ok)

Comment (dyudyunov) [reply common/djangoapps/enrollment/tests/test_views.py]: comment was deleted

File: common/djangoapps/enrollment/views.py (hunk 3 above)
Comment (dyudyunov) [reply common/djangoapps/enrollment/views.py]: thanks, fixed

File: common/djangoapps/enrollment/views.py (hunk 1 above)
Comment (dyudyunov) [reply common/djangoapps/enrollment/views.py]: deleted, thanks )

File: common/djangoapps/enrollment/views.py [hunk 4]
```diff
@@ -649,6 +658,17 @@ def post(self, request):
                     enrollment_attributes=enrollment_attributes
                 )
 
+                course = get_course_by_id(course_id)
```
Comment (GlugovGrGlib) [inline common/djangoapps/enrollment/views.py:661]: Please wrap it in a try/except blovk and log results. Reference: https://github.com/raccoongang/edx-platform/blob/ginkgo-rg/lms/djangoapps/instructor/views/api.py#L545

Comment (dyudyunov) [reply common/djangoapps/enrollment/views.py]: done

**Pull request merged**